labgrid/remote/client.py

```diff
@@ -1122,7 +1127,7 @@ def write_image(self):
         target.activate(drv)
         try:
             drv.write_image(self.args.filename, partition=self.args.partition, skip=self.args.skip,
-                            seek=self.args.seek)
+                            seek=self.args.seek, mode=WRITE_IMAGE_TOOLS[self.args.mode])
```

Comment:
With the change above:
```suggestion
                            seek=self.args.seek, mode=self.args.write_mode)
```